src/OrleansRuntime/MembershipService/SharedSiloStatusListener.cs

```diff
@@ -0,0 +1,40 @@
+﻿
+namespace Orleans.Runtime
+{
+    public struct SiloStatusChange
+    {
+        public SiloStatusChange(SiloAddress updatedSilo, SiloStatus status)
+        {
+            this.UpdatedSilo = updatedSilo;
+            this.Status = status;
+        }
+
+        public SiloAddress UpdatedSilo { get; }
+        public SiloStatus Status { get; }
+    }
+
+    public static class SiloStatusSharedState
+    {
+        public static ISharedState<SiloStatusChange> CreateSiloStatusSharedState(this ISiloStatusOracle oracle)
+        {
+            var currentStatus = new SiloStatusChange(oracle.SiloAddress, oracle.CurrentStatus);
+            var listener = new SharedSiloStatusListener(currentStatus);
+            oracle.SubscribeToSiloStatusEvents(listener);
+            return listener.State;
+        }
+
+        private class SharedSiloStatusListener : SharedStatePublisherBase<SiloStatusChange>, ISiloStatusListener
+        {
+            public SharedState<SiloStatusChange> State => base.currentState;
+
+            public SharedSiloStatusListener(SiloStatusChange startingState) : base(startingState)
+            {
+            }
+
+            public void SiloStatusChangeNotification(SiloAddress updatedSilo, SiloStatus status)
+            {
+                this.Publish(new SiloStatusChange(updatedSilo, status));
```

Comment:
When we get a SiloStatusChangeNotification (on the silo status oracle's thread) all we do is publish the change.  Other threads that want to handle the change must access it via the ISharedState from their own thread.